compiler/rustc_typeck/src/collect.rs

```diff
@@ -1223,6 +1223,36 @@ fn trait_def(tcx: TyCtxt<'_>, def_id: DefId) -> ty::TraitDef {
         ty::trait_def::TraitSpecializationKind::None
     };
     let def_path_hash = tcx.def_path_hash(def_id);
+
+    let must_implement_one_of = tcx
+            .get_attrs(impl_trait_ref.def_id)
+            .iter()
+            .find(|attr| attr.has_name(sym::rustc_must_implement_one_of))
+            .and_then(|attr| match attr.meta_item_list() {
+                None => todo!(), // report error
+                Some(items) if items.len() < 2 => {
+                    tcx.sees
+                        .struct_span_err(
+                            attr.span,
+                            "the `#[rustc_must_implement_one_of]` attribute must be used with at least 2 method names",
+                        )
+                        .emit();
+
+                    None
+                }
+                Some(items) => {
+                    items
+                        .into_iter()
+                        .map(|item| item.ident().ok_or(item.span()))
+                        .collect::<Result<Box<[_]>, _>>().map_err(|span| tcx.sees
+                                            .struct_span_err(
+                                                span,
+                                                "must be an identifier of a method",
+                                            )
+                        .emit()).ok()
```

Comment:
For me, locally `x fmt` doesn't format this part of code. `x test tools/tidy` doesn't error either. This seems like a `rustfmt` bug.